packages/console/src/pages/SignInExperience/components/ManageLanguageModal/CustomLanguageEditor.tsx

```diff
@@ -0,0 +1,146 @@
+import type { LanguageKey } from '@logto/core-kit';
+import resource, { languageCodeAndDisplayNameMappings } from '@logto/phrases-ui';
+import { UiTranslation } from '@logto/phrases-ui/lib/types';
+import { useEffect, useMemo } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import Button from '@/components/Button';
+import useApi, { RequestError } from '@/hooks/use-api';
+import Delete from '@/icons/Delete';
+
+import { createEmptyUiTranslation, flattenObject } from '../../utilities';
+import * as style from './CustomLanguageEditor.module.scss';
+import EditSection from './EditSection';
+import { CustomPhraseResponse } from './types';
+
+type CustomLanguageEditorProps = {
+  selectedLanguageKey: LanguageKey;
+};
+
+const emptyUiTranslation = createEmptyUiTranslation();
+
+const CustomLanguageEditor = ({ selectedLanguageKey }: CustomLanguageEditorProps) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const isBuiltInLanguage = Object.keys(resource).includes(selectedLanguageKey);
+  const translationEntries = useMemo(
+    () => Object.entries(resource[selectedLanguageKey].translation),
+    [selectedLanguageKey]
+  );
+
+  const api = useApi();
+
+  const { data: customPhrase, mutate } = useSWR<CustomPhraseResponse, RequestError>(
+    `/api/custom-phrases/${selectedLanguageKey}`,
+    {
+      shouldRetryOnError: (error: unknown) => {
+        if (error instanceof RequestError) {
+          return error.status !== 404;
+        }
+
+        return true;
+      },
+    }
+  );
+
+  const formMethods = useForm<UiTranslation>({
+    defaultValues: emptyUiTranslation,
```

Comment:
Can we just use undefined as the default Value?

Reply:
If we can use undefined as default value, do we still need this `emptyuiTranslation` object?

Reply:
If we can use undefined as default value, do we still need this `emptyuiTranslation` object?

Reply:
Well, after discussion and some research, the answer is no. The only way to reset a controlled input is to set its default value as a string...